test/queue_list_test.dart

```diff
@@ -258,6 +258,25 @@ void main() {
     expect(queue.cast<Pattern>(), same(queue));
   }, skip: isDart2 ? false : 'Requires a Dart2 runtime');
 
+  test("cast does not throw on mutation when the type is valid", () {
+    var patternQueue = new QueueList<Pattern>()..addAll(['a', 'b']);
+    QueueList<Object> untypedQueue = patternQueue;
```

Comment:
Why this line? Changing the static type has no effect on `cast`, you can just write `patternQueue.cast<String>()` below.

Reply:
Done.